features/step_definitions/tab_steps.rb

```diff
@@ -5,14 +5,15 @@
 
 Then("I should see tabs:") do |table|
   table.rows.each do |title, _|
-    expect(page).to have_css(".tabs .tabs-nav", text: title, visible: true)
+    expect(page).to have_css(".tabs .tabs-nav", text: title, visible: :visible)
   end
 end
 
 Then("I should see tab content {string}") do |string|
-  expect(page).to have_css(".tabs .tabs-content", text: string, visible: true)
+  expect(page).to have_css(".tabs .tabs-content", text: string, visible: :visible)
 end
 
 Then("I should not see tab content {string}") do |string|
+  # FIXME: change to `:all` or `:hidden` according to the expected behavior
   expect(page).to have_css(".tabs .tabs-content", text: string, visible: false)
```

Comment:
Quoting from https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/Capybara/VisibilityMatcher

> `visible: false` does not find just invisible elements, but both visible and invisible elements. For expressiveness and clarity, use one of the # symbol values, `:all`, `:hidden` or `:visible`

@javierjulio if I use `:hidden` here, this spec fails. I can't tell if the expected behavior is `:hidden` or `:all`

```
(::) failed steps (::)

expected to find non-visible css ".tabs .tabs-content" with text "tab 2" but there were no matches. Also found "tab 1\ntab 2", which matched the selector but not all filters.  (RSpec::Expectations::ExpectationNotMetError)
./features/step_definitions/tab_steps.rb:17:in `"I should not see tab content {string}"'
features/show/tabs.feature:34:in `I should not see tab content "tab 2"'

Failing Scenarios:
cucumber features/show/tabs.feature:6 # Scenario: Set a method to be called on the resource as the title
```

Reply:
Yes, this likely due to the JS update which now uses Flowbite instead of jQuery UI tabs on master. You may have created this PR before that and not updated afterwards since I see its behind. Sorry about that. I'm not sure what would be applicable here. I don't recall what happened with jQuery UI tabs, but now each of the tab content areas are defined but hidden using a CSS class. If you rebase this branch, run `bin/rake local server`, login and go to http://localhost:3000/admin/kitchensink you'll see a tabs demo to get an idea what's going. That should help identify what needs to change for the test if anything. Thanks!

Reply:
I do not remember if Cucumber + Capybara in this situation treats `class="hidden"` elements as hidden ones

However, this should fix the spec

```diff
 Then("I should see tab content {string}") do |string|
-  expect(page).to have_css(".tabs .tabs-content", text: string, visible: true)
+  expect(page).to have_css(".tabs .tabs-content :not(.hidden)", text: string)
 end
 
 Then("I should not see tab content {string}") do |string|
-  expect(page).to have_css(".tabs .tabs-content", text: string, visible: false)
+  expect(page).to have_css(".tabs .tabs-content .hidden", text: string)
 end
```

Reply:
Ok great thanks!